hack/make/.binary

```diff
@@ -16,61 +16,88 @@ source "${MAKEDIR}/.go-autogen"
 (
 	export GOGC=${DOCKER_BUILD_GOGC:-1000}
 
-	# for non-sandboxed invocation
-	if ! command -v xx-go > /dev/null 2>&1; then
-		if [ "$(go env GOOS)/$(go env GOARCH)" != "$(go env GOHOSTOS)/$(go env GOHOSTARCH)" ]; then
-			# must be cross-compiling!
-			case "$(go env GOOS)/$(go env GOARCH)" in
-				windows/amd64)
-					export CC="${CC:-x86_64-w64-mingw32-gcc}"
-					export CGO_ENABLED=1
-					;;
-				linux/arm)
-					case "${GOARM}" in
-						5)
-							export CC="${CC:-arm-linux-gnueabi-gcc}"
-							export CGO_ENABLED=1
-							export CGO_CFLAGS="-march=armv5t"
-							export CGO_CXXFLAGS="-march=armv5t"
-							;;
-						6)
-							export CC="${CC:-arm-linux-gnueabi-gcc}"
-							export CGO_ENABLED=1
-							export CGO_CFLAGS="-march=armv6"
-							export CGO_CXXFLAGS="-march=armv6"
-							;;
-						7)
-							export CC="${CC:-arm-linux-gnueabihf-gcc}"
-							export CGO_ENABLED=1
-							export CGO_CFLAGS="-march=armv7-a"
-							export CGO_CXXFLAGS="-march=armv7-a"
-							;;
-						*)
-							export CC="${CC:-arm-linux-gnueabihf-gcc}"
-							export CGO_ENABLED=1
-							;;
-					esac
-					;;
-				linux/arm64)
-					export CC="${CC:-aarch64-linux-gnu-gcc}"
-					export CGO_ENABLED=1
-					;;
-				linux/amd64)
-					export CC="${CC:-x86_64-linux-gnu-gcc}"
-					export CGO_ENABLED=1
-					;;
-				linux/ppc64le)
-					export CC="${CC:-powerpc64le-linux-gnu-gcc}"
-					export CGO_ENABLED=1
-					;;
-				linux/s390x)
-					export CC="${CC:-s390x-linux-gnu-gcc}"
-					export CGO_ENABLED=1
-					;;
-			esac
+	# only necessary for non-sandboxed invocation where TARGETPLATFORM is empty
+	if [ -z "$TARGETPLATFORM" ]; then
+		TARGETPLATFORM="$(go env GOOS)/$(go env GOARCH)"
+		if [ -n "$(go env GOARM)" ]; then
+			TARGETPLATFORM+="/v$(go env GOARM)"
+		elif [ -n "$(go env GOAMD64)" ] && [ "$(go env GOAMD64)" != "v1" ]; then
+			TARGETPLATFORM+="/$(go env GOAMD64)"
 		fi
 	fi
 
+	iscross=0
+	sandboxed=0
```

Comment:
It looks like "sandboxed" really means "is xx in play, overriding the default toolchain names with the cross toolchain, so we don't need to address it by name"? Could we make it more descriptive, e.g. `isxx`? Also, is the availability of `xx-info` enough to detect this? I don't think we should mutate behavior here because a command is in someone's PATH.